From 2e9d85874b2bf0d38d9d3c02699208f9dd1cd136 Mon Sep 17 00:00:00 2001
From: Andrey Gusakov <andrey.gusakov@cogentembedded.com>
Date: Tue, 31 Mar 2015 17:07:56 +0300
Subject: [PATCH] tsdemux: do not discard on discont if PES start

If a discontinuity in the stream is detected, data is discarded until
a new PES starts. If the first packet after the discontinuity is also
the start of a PES, there is no reason to discard the packets.

From 3b149ae7aa36cc989f1ef289d75b7486d93bf425 Mon Sep 17 00:00:00 2001
From: Jesper Larsen <knorr.jesper@gmail.com>
Date: Mon, 29 Sep 2014 10:01:27 +0200
---
 gst/mpegtsdemux/tsdemux.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/gst/mpegtsdemux/tsdemux.c b/gst/mpegtsdemux/tsdemux.c
index c20725c..59b39b4 100644
--- a/gst/mpegtsdemux/tsdemux.c
+++ b/gst/mpegtsdemux/tsdemux.c
@@ -1348,9 +1348,11 @@ gst_ts_demux_queue_data (GstTSDemux * demux, TSDemuxStream * stream,
           (stream->continuity_counter == MAX_CONTINUITY && cc == 0))) {
     GST_LOG ("CONTINUITY: Got expected %d", cc);
   } else {
-    GST_ERROR ("CONTINUITY: Mismatch packet %d, stream %d",
-        cc, stream->continuity_counter);
-    stream->state = PENDING_PACKET_DISCONT;
+    if (stream->state != PENDING_PACKET_EMPTY) {
+      GST_ERROR ("CONTINUITY: Mismatch packet %d, stream %d",
+          cc, stream->continuity_counter);
+      stream->state = PENDING_PACKET_DISCONT;
+    }
   }
   stream->continuity_counter = cc;
 
-- 
1.7.10.4

